File: suhye/BOJ/graph/2606_바이러스.java
import java.util.*;
import java.io.*;

public class Main {

    static int N, M;
    static int count = 0;
    static int[][] graph;
    static boolean[] visit;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());

        graph = new int[N + 1][N + 1];
        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph[a][b] = graph[b][a] = 1;
        }

        visit = new boolean[N + 1];
        dfs(1);
        System.out.println(count - 1); // 1번 컴퓨터 제외해야하므로 -1
    }
    public static void dfs(int id) {
        visit[id] = true;
        count++;
        for(int i = 1; i <= N; i++) {
            if (graph[id][i] == 1 && !visit[i]) {
                dfs(i);
            }
        }
    }
}

File: suhye/BOJ/graph/2606_바이러스.py
import sys

n = int(sys.stdin.readline())
pair = int(sys.stdin.readline())

graph = [[] for _ in range(n+1)]

for _ in range(pair):
    a, b = map(int, sys.stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)

visit = [0] * (n + 1)

cnt = 0


def dfs(start):
    global cnt
    visit[start] = 1
    for i in graph[start]:
        if visit[i] == 0:
            cnt += 1
            dfs(i)

dfs(1)
print(cnt)
